ibazel/lifecycle_hooks/lifecycle_hooks.go

```diff
@@ -0,0 +1,45 @@
+package lifecycle_hooks
+
+import (
+	"bytes"
+	"flag"
+
+	"github.com/bazelbuild/bazel-watcher/ibazel/workspace"
+	"github.com/bazelbuild/bazel-watcher/third_party/bazel/master/src/main/protobuf/blaze_query"
+	"github.com/mattn/go-shellwords"
+)
+
+var runCommandBefore = flag.String("run_command_before", "", "A command to run before each execution")
+
+type LifecycleHooks struct {
+	w workspace.Workspace
+}
+
+func New() *LifecycleHooks {
+	return &LifecycleHooks{
+		w: &workspace.MainWorkspace{},
+	}
+}
+
+func (l *LifecycleHooks) Initialize(info *map[string]string) {}
+
+func (l *LifecycleHooks) TargetDecider(rule *blaze_query.Rule) {}
+
+func (l *LifecycleHooks) ChangeDetected(targets []string, changeType string, change string) {}
+
+func (l *LifecycleHooks) Cleanup() {}
+
+func (l *LifecycleHooks) BeforeCommand(targets []string, command string) {
+	if *runCommandBefore != "" {
+		commandAndArgs, err := shellwords.Parse(*runCommandBefore)
+
+		if err != nil {
+			panic(err)
```

Comment:
I think it's probably reasonable to exit the program on an error here, but WDYT about using ibazel's [`log.Fatalf`](https://godoc.org/github.com/bazelbuild/bazel-watcher/ibazel/log#Fatalf) to provide a user friendly error message that also provides a few examples of what would be acceptable?

Reply:
To be honest, I'm not sure about that. This should return an error if the command is invalid in a shell. I think that should be pretty straightforward for the user to understand what's going on.

Reply:
That's fair, but to me, I think that an unadorned panic with stack trace is definitely not the way to go. People usually file issues against this repo when they get a stacktrace and I don't really want to field those.

Reply:
I see. So we can use the `log.Fatalf`. Only about the `few examples` that I think that it doesn't make sense. What do you think?

Reply:
A few examples might be a bit much. Totally respect that. Let's leave examples out, but write a nice error message still. Does that sound good?